src/OpenTelemetry.AutoInstrumentation.Native/continuous_profiler.cpp

```diff
@@ -218,20 +218,17 @@ int32_t SelectiveSamplingConsumeAndReplaceBuffer(int32_t len, unsigned char* buf
         trace::Logger::Warn("Unexpected 0/null buffer to SelectiveSamplingConsumeAndReplaceBuffer");
         return 0;
     }
-    std::vector<unsigned char>* to_use = nullptr;
-    {
-        std::lock_guard<std::mutex> guard(selective_sampling_buffer_lock);
-        to_use                    = selective_sampling_buffer;
-        selective_sampling_buffer = new std::vector<unsigned char>();
-        selective_sampling_buffer->reserve(kSamplesBufferDefaultSize);
-    }
-    if (to_use == nullptr)
+
+    std::lock_guard<std::mutex> guard(selective_sampling_buffer_lock);
+    const size_t to_use_len = static_cast<int>(std::min(selective_sampling_buffer.size(), static_cast<size_t>(len)));
```

Comment:
static_cast to int is not needed. std::min will return size_t.

Reply:
Removed in https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/4321/commits/134fe29262766d35c7f0dbc409799468c213a6ab